nssp_demo/score_forecast.R

```diff
@@ -0,0 +1,209 @@
+    quantile_scores = quantile_scores
+  ))
+}
+
+
+prep_truth_data <- function(truth_data_path) {
+  dat <- readr::read_tsv(truth_data_path,
+    show_col_types = FALSE
+  ) |>
+    filter(data_type == "test") |>
+    rename(true_value = ed_visits) |>
+    mutate(disease = case_when(
+      disease %in% c("Influenza", "COVID-19") ~ "Disease",
+      disease == "Total" ~ "Total",
+      TRUE ~ NA
+    )) |>
+    filter(!is.na(disease)) |>
+    tidyr::pivot_wider(
+      names_from = "disease",
+      values_from = "true_value"
+    ) |>
+    mutate(prop_disease_ed_visits = Disease / Total) |>
+    tidyr::pivot_longer(
+      c(Disease, Total, prop_disease_ed_visits),
+      names_to = "disease",
+      values_to = "true_value"
+    )
+
+  return(dat)
+}
+
+read_and_score_location <- function(model_run_dir,
+                                    eval_data_filename = "eval_data",
+                                    eval_data_file_ext = "tsv") {
+  message(glue::glue("Scoring {model_run_dir}..."))
+  forecast_path <- fs::path(
+    model_run_dir,
+    "forecast_samples.parquet"
+  )
+  ts_baseline_path <- fs::path(
+    model_run_dir,
+    "baseline_ts_prop_ed_visits_forecast.parquet"
+  )
+  cdc_baseline_path <- fs::path(
+    model_run_dir,
+    "baseline_cdc_prop_ed_visits_forecast.parquet"
+  )
+
+  truth_path <- fs::path(model_run_dir,
+    eval_data_filename,
+    ext = eval_data_file_ext
+  )
+
+  actual_data <- prep_truth_data(truth_path)
+
+  pyrenew <- arrow::read_parquet(forecast_path) |>
+    mutate(model = "pyrenew-hew") |>
+    select(date, .draw, disease, model, .value)
+
+  ts_baseline <- arrow::read_parquet(ts_baseline_path) |>
+    mutate(
+      model = "ts_baseline",
+      disease = "prop_disease_ed_visits"
+    ) |>
+    select(date,
+      .draw,
+      disease,
+      model,
+      .value = prop_disease_ed_visits
+    )
+
+  cdc_baseline <- arrow::read_parquet(cdc_baseline_path) |>
+    mutate(
+      model = "cdc_baseline",
+      disease = "prop_disease_ed_visits"
+    ) |>
+    select(date,
+      disease,
+      quantile_level,
+      .value = baseline_ed_visit_prop_forecast,
+      model
+    )
+
+
+  quantile_forecasts_to_score <- inner_join(
+    cdc_baseline,
+    actual_data,
+    by = c("disease", "date")
+  ) |>
+    filter(disease == "prop_disease_ed_visits")
+
+  sample_forecasts_to_score <- bind_rows(
+    pyrenew,
+    ts_baseline
+  ) |>
+    inner_join(actual_data,
+      by = c("disease", "date")
+    ) |>
+    filter(disease == "prop_disease_ed_visits")
+
+  max_visits <- actual_data |>
+    filter(disease == "Total") |>
+    pull(true_value) |>
+    max()
+
+  scored <- score_single_run(
+    sample_forecasts_to_score,
+    quantile_forecasts_to_score,
+    forecast_unit = c("date", "model"),
+    observed = "true_value",
+    sample_id = ".draw",
+    predicted = ".value",
+    offset = 1 / max_visits
+  )
+
+  saveRDS(scored, fs::path(model_run_dir, "score_table", ext = "rds"))
```

Comment:
consider [`write_rds`](https://readr.tidyverse.org/reference/read_rds.html)

Reply:
Done in 49c3afa